logrotate/src/LogRotateHelper.ipp

```diff
@@ -168,20 +164,22 @@ std::string LogRotateHelper::changeLogFile(const std::string& directory, const s
       file_name = log_prefix_backup_;
    }
 
-   std::string prospect_log = createLogFileName(directory + file_name);
```

Comment:
I think you can keep the createLogFileName but have both these functions inside that main function. that way you get the log file to create in the next line.